segmentation/utils/spatial_analysis_utils.py

```diff
@@ -44,46 +46,68 @@ def compute_close_cell_num(patient_data_markers, label_idx, thresh_vec,
     index [0, 1]).
 
     Args:
+        dist_mat: cells x cells matrix with the euclidian
+            distance between centers of corresponding cells
+        dist_lim: threshold for spatial enrichment distance proximity
+        num: number of markers or cell phenotypes, based on analysis
+        analysis_type: type of analysis, either Cell Label or Threshold
+        cell_label_idx: column of cell labels in Cell Label Analysis
+        patient_data: data for specific patient in Cell Label Analysis
+        cell_pheno_idx: column of cell phenotype id's in Cell Label Analysis
+        pheno_codes: list all the cell phenotypes in Cell Label Analysis
         patient_data_markers: cell expression data of markers
             for the specific point
         label_idx: column of cell labels
         thresh_vec: matrix of thresholds column for markers
-        dist_mat: cells x cells matrix with the euclidian
-            distance between centers of corresponding cells
-        marker_num: number of markers in expresion data
-        dist_lim: threshold for spatial enrichment distance proximity
 
     Returns:
         close_num: marker x marker matrix with counts for cells
             positive for corresponding markers
         marker1_num: list of number of cell labels for marker 1
         marker2_num: list of number of cell labels for marker 2"""
-
     # Create close_num, marker1_num, and marker2_num
-    close_num = np.zeros((marker_num, marker_num), dtype='int')
-    marker1_num = []
-    marker2_num = []
-
-    for j in range(0, marker_num):
-        # Identify cell labels that are positive for respective markers
-        marker1_thresh = thresh_vec.iloc[j]
-        marker1posinds = patient_data_markers[patient_data_markers.columns[j]] > marker1_thresh
-        marker1poslabels = label_idx[marker1posinds]
-        marker1_num.append(sum(marker1posinds))
-        for k in range(0, marker_num):
-            # Identify cell labels that are positive for the kth marker
-            marker2_thresh = thresh_vec.iloc[k]
-            marker2posinds = patient_data_markers[patient_data_markers.columns[k]] > marker2_thresh
-            marker2poslabels = label_idx[marker2posinds]
-            marker2_num.append(sum(marker2posinds))
+    close_num = np.zeros((num, num), dtype='int')
+    mark1_num = []
+    mark2_num = []
+
+    for j in range(0, num):
+        # Identify cell labels that are positive for respective markers, based on type of analysis
+        mark1poslabels = None
+        if analysis_type == "Cell Label":
```

Comment:
Let's break this out into a separate function, get_cell_labels, that does this logic. It should take in analysis_type, and the appropriate data, and return the positive labels.